contracts/RewardsPoolBase.sol

```diff
@@ -323,20 +344,27 @@ contract RewardsPoolBase is Ownable {
         }
 
         uint256 applicableTimestamp = (_currentTimestamp < endTimestamp) ? _currentTimestamp : endTimestamp;
-
         uint256 secondsSinceLastReward = applicableTimestamp - lastRewardTimestamp;
+        uint256 rewardsTokensLength = rewardsTokens.length;
 
         if (secondsSinceLastReward == 0) {
             return;
         }
 
         if (totalStaked == 0) {
+            uint256[] memory _rewardPerSecond = new uint256[](rewardsTokensLength);
```

Comment:
This breaks in the following scenario:

You have a 30 day campaign, somebody stakes at 0 days, then withdraws at 5 days. So now there is 0 totalStaked again. Then at 10 days he stakes again, so now it will take all rewards and spread it over the last 20 days. But actually some rewards were already used in the first 5 days. No there is not enough rewards, and it will break.

We need to make sure it only can do this the first time that totalStaked is 0